crates/datafusion_ext/Cargo.toml

```diff
@@ -0,0 +1,15 @@
+[package]
+name = "datafusion_ext"
+description = "Shared datafusion extensions"
+version = { workspace = true }
+edition = { workspace = true }
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+datafusion = { workspace = true }
+uuid = { version = "1.4.0", features = ["v4", "fast-rng", "macro-diagnostics"] }
+regex = "1.8"
+once_cell = "1.18.0"
+tracing = "0.1"
+thiserror.workspace = true
```

Comment:
i'll take a look & see if we can move this into another crate.

Reply:
so i was examining the dependencies and we _could_ definitely move it to `sqlbuiltins`, but i don't really think of the session variables as sql builtins & have really nothing to do with sql, but with the session context. So while we _could_, i don't think we **should** move it into there. 


one suggestion

rename `datafusion_planner` to datafusion_ext` & move the sessionvar stuff into there. 
This would require moving a few other components from sqlbuiltins into datafusion_ext as well to break the dependency. 


alternatively, we could just move it in to `sqlbuiltins` as is. 

I'm in favor of moving it into a datafusion extension crate, but i'd like to get your opinion before moving ahead. 

@scsmithr

Reply:
> rename `datafusion_planner` to datafusion_ext` & move the sessionvar stuff into there. This would require moving a few other components from sqlbuiltins into datafusion_ext as well to break the dependency.
> 
> alternatively, we could just move it in to `sqlbuiltins` as is.
> 
> I'm in favor of moving it into a datafusion extension crate, but i'd like to get your opinion before moving ahead.

I'm down with moving the datafusion planner stuff.